src/ipahealthcheck/dogtag/ca.py

```diff
@@ -102,10 +103,40 @@ def check(self):
             logger.debug('CA is not configured, skipping connectivity check')
             return
 
+        config = api.Command.config_show()
+        subject_base = config['result']['ipacertificatesubjectbase'][0]
+        ipa_subject = DN(('CN', 'Certificate Authority'), subject_base)
+        try:
+            certs = x509.load_certificate_list_from_file(paths.IPA_CA_CRT)
+        except Exception as e:
+            yield Result(self, constants.ERROR,
+                         key='ipa_ca_crt_file_missing',
+                         path=paths.IPA_CA_CRT,
+                         exception=str(e),
+                         msg='The IPA CA cert file {path} could not be'
+                             'opened: {exception}')
+            return
+
+        found = False
+        for cert in certs:
+            if DN(cert.subject) == ipa_subject:
+                found = True
+                break
+
+        if not found:
+            yield Result(self, constants.ERROR,
+                         key='ipa_ca_cert_not_found',
+                         subject=ipa_subject,
```

Comment:
ipa_subject needs to be cast to a str, otherwise ipa-healthcheck outputs: `Output raised TypeError: Object of type DN is not JSON serializable`

Reply:
Thanks, nice catch. It also means I'm missing a test case.

Reply:
Updated patch to fix the DN issue and include several new tests for better coverage.